junit-platform-console/src/main/java/org/junit/platform/console/tasks/DiffPrinter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.console.tasks;
+
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+
+import com.github.difflib.text.DiffRow;
+import com.github.difflib.text.DiffRowGenerator;
+
+/**
+ * class provide access to printDiff function
+ */
+class DiffPrinter {
+	//print the difference of two print to out
+	static void printDiff(PrintWriter out, String expected, String actual) {
+		DiffRowGenerator generator = DiffRowGenerator.create().showInlineDiffs(true).inlineDiffByWord(true).oldTag(
```

Comment:
I'm not sure about `inlineDiffByWord`. If the string is short, doing the inline diff by character might be more helpful. WDYT?

Reply:
`inlineDiffByWord` makes sense when comparing two words or two sentences, which is the most use case I think?
Without `inlineDiffByWord`, the output will exclude the individual identical letter from the markdown.
For example, ~~T~~e~~st~~ AND **L**e**ft**.
No `inlineDiffByWord` is better when comparing two long strings without space in between. In that case, writing a small algorithm to return the index of different characters is more efficient.

Reply:
Maybe we can apply a heuristic: If one of the strings contains whitespace, we'd enable `inlineDiffByWord`; otherwise, we'd disable it.